frontend/src/employee-frontend/components/messages/MessagesPage.tsx

```diff
@@ -52,24 +60,37 @@ export default React.memo(function MessagesPage() {
   useEffect(() => refreshMessages(), [refreshMessages])
   const [sending, setSending] = useState(false)
 
-  // pre-select first account on page load and on unit change
+  // select first account if no account is selected
   useEffect(() => {
-    if (!accounts.isSuccess) {
+    if (!accounts.isSuccess || selectedAccount) {
       return
     }
-    const { value: data } = accounts
-    const unitSelectionChange =
-      selectedAccount &&
-      !data.find((acc) => acc.account.id === selectedAccount.account.id)
-    if ((!selectedAccount || unitSelectionChange) && data.length > 0) {
-      setSelectedAccount({
-        view: 'RECEIVED',
-        account:
-          data.find((a) => a.account.type === 'PERSONAL')?.account ||
-          data[0].account
+
+    if (municipalAccount) {
+      return selectAccount({
+        view: municipalMessageBoxes[0],
+        account: municipalAccount.account,
+        unitId: null
+      })
+    }
+
+    if (personalAccount) {
+      return selectAccount({
+        view: personalMessageBoxes[0],
+        account: personalAccount.account,
+        unitId: null
+      })
+    }
+
+    const firstGroupAccount = groupAccounts[0]
+    if (firstGroupAccount) {
+      return selectAccount({
+        view: groupMessageBoxes[0],
+        account: firstGroupAccount.account,
+        unitId: firstGroupAccount.daycareGroup.unitId
       })
     }
-  }, [accounts, setSelectedAccount, selectedAccount])
+  }, [accounts, groupAccounts, municipalAccount, personalAccount, selectAccount, selectedAccount])
```

Comment:
Would it be cleaner if this effect lived in the context provider instead of inside this component? 🤔 

Also: Would it be possible to use `renderResult` a few lines down where the JSX is rendered?

Reply:
Do you mean this part?

```
{showEditor && accounts.isSuccess && receivers && selectedAccount && (
  <MessageEditor ... />
)}
```

Reply:
Do you mean this part?

```
{showEditor && accounts.isSuccess && receivers && selectedAccount && (
  <MessageEditor ... />
)}